extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -94,6 +127,30 @@ void transformWithSpan(
         }));
     }
 
+    @BuildStep
+    void transformSpanAttribute(
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+        // Transform deprecated annotation into new one
+        Collection<AnnotationInstance> legacySpanAttribute = combinedIndex.getIndex().getAnnotations(DotName.createSimple(
+                "io.opentelemetry.extension.annotations.SpanAttribute"));
+
+        // Transform deprecated annotation into new one
+        annotationsTransformer.produce((new AnnotationsTransformerBuildItem(context -> {
+            AnnotationTarget target = context.getTarget();
+            for (AnnotationInstance legacyAnnotation : legacySpanAttribute) {
+                if (target.equals(legacyAnnotation.target())) {
```

Comment:
`AnnotationTarget` as such doesn't implement `equals` method so I am not sure this will always do what you want it it...

Reply:
this method was not really needed. Dropped it